packages/wp-story-editor/src/components/fontCheck/index.js

```diff
@@ -58,6 +62,22 @@ export const FontCheck = () => {
       const inUseFonts = getInUseFontsForPages(storyPages);
       const allFonts = await getFonts({ include: inUseFonts.join(',') });
 
+      if (!fontsChecked) {
+        inUseFonts.map((fontFamily) => {
+          const matchedFont = allFonts.find(
+            (font) => font.family === fontFamily
+          );
+          if (matchedFont) {
+            // ensure `in use` font element uses the latest metrics, weights etc...
+            updateElementsByFontFamily({
+              family: fontFamily,
+              properties: { font: matchedFont },
+            });
+          }
+        });
```

Comment:
Nit: a `for .. of` loop is more suitable here than `.map()` as we don't use the return values here.